worker/uniter/runner/context/contextfactory.go

```diff
@@ -65,12 +64,8 @@ type StorageContextAccessor interface {
 
 type SecretsAccessor interface {
 	SecretIds() (map[*secrets.URI]string, error)
-	Create(cfg *secrets.SecretConfig, ownerTag names.Tag, value secrets.SecretValue) (string, error)
-	Update(uri string, cfg *secrets.SecretConfig, value secrets.SecretValue) error
-	Remove(uri string) error
-	GetValue(uri, label string, update, peek bool) (secrets.SecretValue, error)
-	Grant(uri string, p *secretsmanager.SecretRevokeGrantArgs) error
-	Revoke(uri string, p *secretsmanager.SecretRevokeGrantArgs) error
+	Create(cfg *secrets.SecretConfig, ownerTag names.Tag, value secrets.SecretValue) (*secrets.URI, error)
```

Comment:
Will Create be moved into the Commit phase of hook execution like we had discussed?

Reply:
Yes, need to expose the means to get the URI ahead of time first